dynamically_sized_list.c

```diff
@@ -0,0 +1,243 @@
+#include "stdlib.h"
+#include "string.h"
+#include "stdio.h"
+#include "dynamically_sized_list.h"
+
+void readStringFromConsole(char** str_)
+{
+	/*	
+		this function allocate dynamic memory for sentence in stdin buffer (until '\n' is reached)
+		and assign this string to str_
+	*/
+	if (!str_) return;
+	int reallocSize = REALLOC_SIZE, // value of memory to be added every reallocation
+		offset = 0; //current position
+	char* str = NULL;
+	do {
+		str = (char*)malloc(reallocSize * sizeof(char));
+	} while (!str);
+	str[0] = '\0';
+	char* ptr2 = str;
+	printf("Enter your string. Press '\\n' to end your string\n");
+	char ch;
+	while (1)
+	{
+		if (offset != 0 && (offset % reallocSize) == 0) //check for available allocated memory
+		{
+			reallocSize += reallocSize;
+			do {
+				ptr2 = (char*)realloc(ptr2, reallocSize);

```

Comment:
realloc() possible leak: when realloc() fails in allocating memory, original pointer 'ptr2' is lost. Consider assigning realloc() to a temporary pointer.